### PR TITLE
Jspss93094 hw3 attempt1

### PR DESCRIPTION
**Pull request by jspss93094**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

**Comment by jspss93094** [comment]
Excuse me. I found that the validate.py used 500*500 matrix for testing, and it is different with the assignment description. So I  can not pass it. Should I make it pass in that condition?

**Comment by tigercosmos** [comment]
@jspss93094  please let it pass the CI.

**Comment by jspss93094** [comment]
@tigercosmos  I have updated it. Thanks :)  And I actually did not change files other than my own folder, but it shows in Files changed. Did I do something wrong?

**Comment by tigercosmos** [comment]
try

```
git remote add upstream https://github.com/yungyuc/nsdhw_20au
git fetch upstream
git rebase upstream/hw3
```

**Comment by jspss93094** [comment]
@tigercosmos Ok. Thanks :)

File: hw3/jspss93094/matrix.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    {
+        reset_buffer(0, 0);
+        std::swap(m_nrow, other.m_nrow);
+        std::swap(m_ncol, other.m_ncol);
+        std::swap(m_buffer, other.m_buffer);
+    }
+
+    ~Matrix()
+    {
+        reset_buffer(0, 0);
```
Comment (tigercosmos) [inline hw3/jspss93094/matrix.cpp:33]: no need this line.

**Pull request merged**